hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1227,19 +1235,25 @@ def defer_recursion(thing, producer):
             hints = get_type_hints(thing)
             params = get_signature(thing).parameters
         except Exception:
-            return builds(thing)
+            params = {}
         kwargs = {}
         for k, p in params.items():
             if (
                 k in hints
                 and k != "return"
-                and p.default is not Parameter.empty
```

Comment:
Finally! The point is that without removing this check there are far too many false positives.

Reply:
Note that this makes the result more explicit, which changes ghostwriter output. Where you'd earlier see `builds(CustomClass)`, you now get `builds(CustomClass, number=st.integers())`.

IMO this is an improvement, but I realize opinions may differ. It's certainly more consistent with the has-default-value case.

Reply:
I agree that the new, more explicit repr is better.